tests/cases/conformance/salsa/typeFromPropertyAssignment16.ts

```diff
@@ -0,0 +1,19 @@
+// @noEmit: true
+// @allowJs: true
+// @checkJs: true
+// @Filename: a.js
+var Outer = {};
+
+Outer.Inner = function () {}
+Outer.Inner.prototype = {
+    x: 1,
+    m() { }
```

Comment:
We should also check/enforce that a `__proto__` member in the RHS acts as a base type for the class type (this is also present in the devtools codebase).

Reply:
Ah, Annex B... 😭

Reply:
`__proto__` was sensible ish in a pre-`Object.setPrototypeOf` world. :P

Reply:
But there was `Object.create` :(

Reply:
I’ll add it to the Big List of JavaScript We Could Do Better. I don’t wa t to put it in this PR though — I have t touched the inheritance code for a while.

Reply:
K.